compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2104,12 +2104,38 @@ class Typer extends Namer
     lazy val annotCtx = annotContext(mdef, sym)
     // necessary in order to mark the typed ahead annotations as definitely typed:
     for (annot <- mdef.mods.annotations)
-      checkAnnotApplicable(typedAnnotation(annot)(using annotCtx), sym)
+      val annot1 = typedAnnotation(annot)(using annotCtx)
+      checkAnnotApplicable(annot1, sym)
+      if (Annotations.annotClass(annot1) == defn.NowarnAnnot)
+        registerNowarn(annot1, mdef)
   }
 
   def typedAnnotation(annot: untpd.Tree)(using Context): Tree =
     checkAnnotArgs(typed(annot, defn.AnnotationClass.typeRef))
 
+  def registerNowarn(tree: Tree, mdef: untpd.Tree)(using Context): Unit =
+    val annot = Annotations.Annotation(tree)
+    def argPos = annot.argument(0).getOrElse(tree).sourcePos
+    val filters =
+      if annot.arguments.isEmpty then List(MessageFilter.Any)
+      else annot.argumentConstantString(0) match {
+        case None => annot.argument(0) match {
+          case Some(t: Select) if t.name.toString == "$lessinit$greater$default$1" => List(MessageFilter.Any)
```

Comment:
you could check `t.name.is(DefaultGetterName)` but I don't know if that works for things coming from scala 2, otherwise you could add the symbol for this getter to Definitions.scala and check if `t.symbol` is equal to it.